### PR TITLE
Issue#822 - API Gateway Async Lambda Invocation

### LINKED ISSUE #822: API Gateway Async Lambda Invocation
### Describe the feature

This feature allows users to build an API Gateway with an asynchronous Lambda integration, in addition to a DynamoDB integration. The POST method submits an async request to Lambda, and returns a jobID to the user.

The user can then use the GET method to query the jobStatus for that jobId. The GET method relies on direct DynamoDB integration with API Gateway.

### Use Case

Giving users the ability to receive a jobId for an async invocation, thus allowing them to check the jobStatus for that jobId.

### Proposed Solution

I am using API Gateway context to return a requestId upon a POST request. The Lambda function stores this requestId in the DynamoDB table.

The GET method queries the DynamoDB table for that requestId, and returns the value of the status attribute.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Language

Typescript

### PR DESCRIPTION
**Pull request by jfan9**
Add a cdk example of Api Gateway asynchronously invoking a lambda function with `POST` request to process data in the backend and return a `jobId`, in addition to query the jobId status from a DynamoDB table with `GET` request. The POST method submits an async request to Lambda, and returns a `jobID` to the user.

Fixes #822

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: typescript/api-gateway-async-lambda-invocation/lib/api-gateway-async-lambda-invocation-stack.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    // Create a Log Group for API Gateway logs
+    const fnLogGroup = new LogGroup(this, `${props.prefix}-fn-log-group`, {
+      retention: RetentionDays.ONE_WEEK, // Customize the retention period as needed
+    });
+
+    //  create a lambda function
+    const jobHandler = new Function(this, `${props.prefix}-fn`, {
+      runtime: Runtime.NODEJS_20_X,
+      handler: 'job_handler.handler',
+      code: Code.fromAsset(path.join(__dirname, '../assets/lambda-functions')),
```
Comment (kaiz-io) [inline typescript/api-gateway-async-lambda-invocation/lib/api-gateway-async-lambda-invocation-stack.ts:34]: Missing assets folder with the code for the lambda function.

**Review by kaiz-io — CHANGES_REQUESTED** [review]
Just missing the lambda code otherwise looks good and thanks for the contribution.

File: typescript/api-gateway-async-lambda-invocation/lib/api-gateway-async-lambda-invocation-stack.ts (hunk 1 above)
Comment (jfan9) [reply typescript/api-gateway-async-lambda-invocation/lib/api-gateway-async-lambda-invocation-stack.ts:34]: [job_handler.js](https://github.com/aws-samples/aws-cdk-examples/pull/1089/commits/26b5bc3ce716d8cddc785fcaee65ec28091d038c) has been added.

**Review by kaiz-io — APPROVED** [review]
:+1:

**Pull request merged**